tls/s2n_security_policies.c

```diff
@@ -1205,147 +1093,147 @@ const struct s2n_security_policy security_policy_null = {
+    { .version = "20210825_gcm", .security_policy = &security_policy_20210825_gcm, .ecc_extension_required = 0 },
+    { .version = "20170405", .security_policy = &security_policy_20170405, .ecc_extension_required = 0 },
+    { .version = "20170405_gcm", .security_policy = &security_policy_20170405_gcm, .ecc_extension_required = 0 },
+    { .version = "20170718", .security_policy = &security_policy_20170718, .ecc_extension_required = 0 },
+    { .version = "20170718_gcm", .security_policy = &security_policy_20170718_gcm, .ecc_extension_required = 0 },
+    { .version = "20190120", .security_policy = &security_policy_20190120, .ecc_extension_required = 0 },
+    { .version = "20190121", .security_policy = &security_policy_20190121, .ecc_extension_required = 0 },
+    { .version = "20190122", .security_policy = &security_policy_20190122, .ecc_extension_required = 0 },
+    { .version = "20190801", .security_policy = &security_policy_20190801, .ecc_extension_required = 0 },
+    { .version = "20190802", .security_policy = &security_policy_20190802, .ecc_extension_required = 0 },
+    { .version = "20200207", .security_policy = &security_policy_test_all_tls13, .ecc_extension_required = 0 },
+    { .version = "20201021", .security_policy = &security_policy_20201021, .ecc_extension_required = 0 },
+    { .version = "20210816", .security_policy = &security_policy_20210816, .ecc_extension_required = 0 },
+    { .version = "20210816_GCM", .security_policy = &security_policy_20210816_gcm, .ecc_extension_required = 0 },
+    { .version = "20240603", .security_policy = &security_policy_20240603, .ecc_extension_required = 0 },
+    { .version = "rfc9151", .security_policy = &security_policy_rfc9151, .ecc_extension_required = 0 },
+    { .version = "test_all", .security_policy = &security_policy_test_all, .ecc_extension_required = 0 },
+    { .version = "test_all_fips", .security_policy = &security_policy_test_all_fips, .ecc_extension_required = 0 },
+    { .version = "test_all_ecdsa", .security_policy = &security_policy_test_all_ecdsa, .ecc_extension_required = 0 },
+    { .version = "test_all_rsa_kex", .security_policy = &security_policy_test_all_rsa_kex, .ecc_extension_required = 0 },
+    { .version = "test_ecdsa_priority", .security_policy = &security_policy_test_ecdsa_priority, .ecc_extension_required = 0 },
+    { .version = "test_all_tls12", .security_policy = &security_policy_test_all_tls12, .ecc_extension_required = 0 },
+    { .version = "test_all_tls13", .security_policy = &security_policy_test_all_tls13, .ecc_extension_required = 0 },
+    { .version = "null", .security_policy = &security_policy_null, .ecc_extension_required = 0 },
+    { .version = NULL, .security_policy = NULL, .ecc_extension_required = 0 }
 };
 
 const char *deprecated_security_policies[] = {
+    "KMS-PQ-TLS-1-0-2019-06",
+    "KMS-PQ-TLS-1-0-2020-02",
+    "KMS-PQ-TLS-1-0-2020-07",
```

Comment:
Your changes to the policy list are very hard / impossible to visually verify. Could you maybe run a manual test to confirm that s2n_array_len(security_policy_selection) + s2n_array_len(deprecated_security_policies) is the same before and after your change? Like maybe add some logic to a test or other binary that can report that number, and then run that test with and without your change.

I'm not sure we could add a permanent, committed test without requiring that it be updated every time someone adds a new policy :/

It might be even better if you can confirm no changes to the names. Maybe the test/binary could write the names to a file instead of just reporting a number, and then you can diff the files with and without your change.